griptape/tools/griptape_cloud_tool/tool.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from types import MethodType
+from typing import Any, Callable
+from urllib.parse import urljoin
+
+import requests
+from attrs import define, field
+from schema import Literal, Optional, Schema
+
+from griptape.artifacts.base_artifact import BaseArtifact
+from griptape.tools.base_griptape_cloud_tool import BaseGriptapeCloudTool
+from griptape.utils.decorators import activity
+
+
+@define()
+class GriptapeCloudToolTool(BaseGriptapeCloudTool):
+    """Runs a Griptape Cloud hosted Tool.
+
+    Attributes:
+        tool_id: The ID of the tool to run.
+    """
+
+    tool_id: str = field(kw_only=True)
+
+    def __attrs_post_init__(self) -> None:
+        self.__init_activities()
+
+    def __init_activities(self) -> None:
+        schema = self.__get_schema()
+        tool_name, activity_schemas = self.__parse_schema(schema)
+
+        if self.name == self.__class__.__name__:
+            self.name = tool_name
+
+        for activity_name, (description, activity_schema) in activity_schemas.items():
+            activity_handler = self.__create_activity_handler(activity_name, description, activity_schema)
+
+            setattr(self, activity_name, MethodType(activity_handler, self))
+
+    def __get_schema(self) -> dict:
```

Comment:
Why are we using leading double underscores? Are we trying to prevent users from overriding this method in subclasses?

As I understand it, adding leading double underscores will cause the class member's name to be mangled. In this case it will change `__get_schema` to `_GriptapeCloudToolTool__get_schema`. So my question is why?


> Or is this just related to `attrs`?

Reply:
I guess I was feeling overly private about these fields. Marking them as protected is more consistent with our usual behavior, I'll do that.